pkg/k8s/k8s_pod.go

```diff
@@ -247,41 +262,83 @@ func buildEnv(envMap map[string]string) []v1.EnvVar {
 
 // buildPodVolumes generates a volume configuration for a pod based on the given name.
 // If the volumes amount is zero, returns an empty slice.
-func buildPodVolumes(name string, volumesAmount int) ([]v1.Volume, error) {
-	if volumesAmount == 0 {
+func buildPodVolumes(name string, volumesAmount, filesAmount int) ([]v1.Volume, error) {
+	// return empty slice if no volumes or files are specified
+	if volumesAmount == 0 && filesAmount == 0 {
 		return []v1.Volume{}, nil
 	}
 
-	podVolume := v1.Volume{
-		Name: name,
-		VolumeSource: v1.VolumeSource{
-			PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
-				ClaimName: name,
+	var podVolumes []v1.Volume
+
+	if volumesAmount != 0 {
+		podVolume := v1.Volume{
+			Name: name,
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					ClaimName: name,
+				},
 			},
-		},
+		}
+
+		podVolumes = append(podVolumes, podVolume)
 	}
 
-	return []v1.Volume{podVolume}, nil
+	defaultMode := int32(0777)
```

Comment:
`0777` 🥲 is it need it?, we're currently using [0550](https://github.com/celestiaorg/celestia-stack-internal/blob/main/data-availability/base/statefulset.yaml#L142) here